### PR TITLE
rmf_audio: enhanced error code Macro implementation

### PR DESCRIPTION
**Pull request by thanushreevani**

### CODE REVIEW

**Comment by hari22yuva** [comment]
hi @thanushreevani ,
write a description what this PR is

**Review by hari22yuva — CHANGES_REQUESTED** [review]
if RMF_INVALID_STATE,RMF_INVALID_HANDLE are enhanced error code what is normal codes, that we need to guess and add else case not disabling the calling function

File: src/test_l1_rmfAudioCapture.c [hunk 1]
```diff
@@ -242,11 +242,13 @@ void test_l1_rmfAudioCapture_negative_RMF_AudioCapture_Open_Type_primary (void)
 		UT_FAIL_FATAL("Aborting test - unable to open capture.");
 	}
 
+    #ifdef ENABLE_ENHANCED_ERROR_CODE
 	result = RMF_AudioCapture_Open_Type(&handle, RMF_AC_TYPE_PRIMARY);
 	UT_ASSERT_EQUAL(result, RMF_INVALID_STATE);
 	if(RMF_INVALID_STATE != result){
 		UT_FAIL("Failed to reject a second primary capture request.");
 	}
+	#endif
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_rmfAudioCapture.c:258]: Done

File: src/test_l1_rmfAudioCapture.c [hunk 2]
```diff
@@ -243,10 +263,10 @@ void test_l1_rmfAudioCapture_negative_RMF_AudioCapture_Open_Type_primary (void)
 	}
 
 	result = RMF_AudioCapture_Open_Type(&handle, RMF_AC_TYPE_PRIMARY);
-	UT_ASSERT_EQUAL(result, RMF_INVALID_STATE);
-	if(RMF_INVALID_STATE != result){
+	CHECK_FOR_EXTENDED_ERROR_CODE(result, RMF_INVALID_STATE, TEST_RMF_OSAL_EBUSY);
+	/*if(RMF_INVALID_STATE != result){
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_rmfAudioCapture.c]: Done

**Review by srinivasgtl — APPROVED** [review]
approved

**Review by hari22yuva — APPROVED** [review]
changes looks ok

**Pull request merged**